easybuild/tools/modules.py

```diff
@@ -1413,6 +1417,15 @@ def get_setenv_value_from_modulefile(self, mod_name, var_name):
 
         return value
 
+    def update(self):
+        """Update after new modules were added."""
+
+        if build_option('update_modules_tool_cache') and self.supports_module_cache:
```

Comment:
No need to introduce `self.supports_module_cache` or `REQ_VERSION_MODULE_CACHE`  I think, just include the version check here directly?

The `self.supports_*` mechanism is mainly useful when `module_generator` needs to figure out whether the active modules tool supports something, but there's no need for that indirection here...

```python
        if build_option('update_modules_tool_cache') and version >= LooseVersion('5.3.0'):
```

Reply:
Thanks for your feedback. I have revised the code to take your suggestion into account.